GradeAnalytics/src/DataSet.java

```diff
@@ -634,6 +636,97 @@ public String getLastError()
 	{
 		return errorLog.get(errorLog.size() - 1);
 	}
+	
+	/**
+	 * Generates a report of all of the data
+	 * @return A string indicating the success
+	 * of the function.
+	 */
+	public String generateReport()
+	{
+		String message;
+		File report = new File("report.txt");
+		if (data.size() == 0)
+		{
+			message = "There is no data to generate a report";
+		}
+		else
+		{
+			try
+			{
+				report.createNewFile();
+				PrintStream writer = new PrintStream(report);
+				String toAppend;
+				// Max
+				toAppend = getMaxBoundary();
+				writer.println(toAppend);
+				// Min
+				toAppend = getMinBoundary();
+				writer.println(toAppend);
+				// The Dataset
+				
+				// Simple Stats
+				// Count, Max, Min, Mean, Median, Mode
+				toAppend = "Data Count: " + getDataCount();
+				writer.println(toAppend);
+				toAppend = "Max: " + getMax();
+				writer.println(toAppend);
+				toAppend = "Min: " + getMin();
+				writer.println(toAppend);
+				toAppend = getMean();
+				writer.println(toAppend);
+				toAppend = getMedian();				
+				writer.println(toAppend);
+				toAppend = getMode();				
+				writer.println(toAppend);
+				// Distribution
+				
+				// Error Log/Graph?
```

Comment:
Not sure if we should include the error log or graph in our report implementation

Reply:
Confirmed by TA we need a graph in our report

Reply:
Confirmed by TA we need to include the error log